### PR TITLE
Kubernetes UI with Dashboard component

### PR DESCRIPTION
**Pull request by ohohoreilly**
Adding the Kubernetes UI dashboard component.

### CODE REVIEW

**Comment by preillyme** [comment]
pr

@lavalamp I'm running in to an issue with cAdvisor and accessing the data over SSL. I need to avoid the ```Mixed Content: The page at 'https://104.154.67.146/static/app/#/dashboard/' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://104.197.36.27:4194/api/v1.0/containers/'. This request has been blocked; the content must be served over HTTPS.```

**Comment by preillyme** [comment]
I tried something like:
```https://104.197.21.127/api/v1beta3/proxy/nodes/kubernetes-minion-7jpp.c.shared-kraken.internal:4194/api/v1.0/containers``` but I receive the following error:
```
{
  "kind": "Status",
  "apiVersion": "v1beta3",
  "metadata": {},
  "status": "Failure",
  "message": "minion \"kubernetes-minion-7jpp.c.shared-kraken.internal:4194\" not found",
  "reason": "NotFound",
  "details": {
    "id": "kubernetes-minion-7jpp.c.shared-kraken.internal:4194",
    "kind": "minion"
  },
  "code": 404
}
```

**Comment by preillyme** [comment]
The code in the Web UI that is setting up the request is found in: https://github.com/kubernetes-ui/kubernetes/blob/kubernetes-ui-dashboard/www/master/shared/js/modules/services/cAdvisor.js#L10

**Comment by preillyme** [comment]
@lavalamp was thinking that doing something like: ```strings.Contains("kubernetes-minion-7jpp.c.shared-kraken.internal:4194", ":")``` and ``` host, port, _ := net.SplitHostPort(u.Host)``` with the proxy in https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/apiserver/proxy.go#L53 so the goal is for something like https://104.197.21.127/api/v1beta3/proxy/nodes/kubernetes-minion-7jpp.c.shared-kraken.internal:4194/api/v1.1/containers/ to work as it does directly at http://104.197.33.84:4194/api/v1.1/containers/

**Comment by lavalamp** [comment]
Please see #7073

**Comment by preillyme** [comment]
@lavalamp I'm trying a build with that fix now.

**Comment by preillyme** [comment]
pr 1

I'm not really sure why the kubernetes.svg isn't appearing. It's accessible on my test cluster at: https://130.211.155.227/static/app/assets/img/kubernetes.svg

**Comment by preillyme** [comment]
@roberthbailey any ideas about why the logo wouldn't be appearing?

**Comment by preillyme** [comment]
Looks like the SVG issue is now fixed: 
 with the changes in https://github.com/GoogleCloudPlatform/kubernetes/pull/7083

File: www/README.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+To write new Karma tests:
+* For testing each components, write test files (`*.spec.js`) under the
+corresponding `www/master/components/**/test/modules/` directory.
+* For testing the chrome and the framework, write test files
+  (*.spec.js) under the `www/master/test/modules/*` directory.
+
+#### End-to-end testing via Protractor
+To run the existing Protractor tests:
+* Install the CLIs: `sudo npm install -g protractor`.
+* Start the webdriver server: `sudo webdriver-manager start`
```
Comment (lavalamp) [inline www/README.md]: What do I need to install to get this command?

Comment (preillyme) [inline www/README.md]: @lavalamp Unittests via Karma

To run the existing Karma tests:

Install the Karma CLI: sudo npm install -g karma-cli (it needs to be installed globally, hence the sudo may be needed). Note that the other Karma packages (such as karma, karma-jasmine, and karma-chrome-launcher should be automatically installed when running npm start).
Go to the www/master directory, and run karma start karma.conf.js. The Karma configuration is defined in karma.config.js. The console should show the test results.

**Comment by lavalamp** [comment]
There are a number of things in this change that I don't think are up to production grade yet, however instead of bogging this down and asking for fixes for all of them, I've filed #7122. Please tag me in future UI PRs.

**Comment by preillyme** [comment]
:+1:

**Comment by jackgr** [comment]
:+1:

**Pull request merged**